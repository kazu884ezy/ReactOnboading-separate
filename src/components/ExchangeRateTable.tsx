import React from 'react';
import { TableHead, TableBody, TableRow, TableCell} from "@material-ui/core";
import StripeTable from "./StripeTable";

// const staticData = [
//     ['USD', '1.5'],
//     ['EUR', '2.0'],
//     ['AUD', '1.1'],
//     ['JPY', '0.03'],
// ]

interface DatePickeExchangeRateTableProps {
    exchangeRates : string[][]
}

const ExchangeRateTable = ({exchangeRates} : DatePickeExchangeRateTableProps) => {
    const renderResults = (dataArray: string[][]) => {
        return dataArray.map(data => {
            return (
                <TableRow key={data[0]}>
                    <TableCell>{data[0]}</TableCell>
                    <TableCell>{(+data[1]).toFixed(2)}</TableCell>
                </TableRow>
            );
        })
    }

    return (
        <StripeTable>
            <TableHead >
                <TableRow>
                    <TableCell>
                        Currency
                    </TableCell>
                    <TableCell>
                        Value
                    </TableCell>
                </TableRow>
            </TableHead>
            <TableBody>
                {renderResults(exchangeRates)}
            </TableBody>
        </StripeTable>
    );
};

export default ExchangeRateTable;
