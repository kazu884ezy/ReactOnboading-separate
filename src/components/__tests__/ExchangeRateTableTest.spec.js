import React from 'react';
import { render, screen} from '@testing-library/react';
import ExchangeRateTable from "../ExchangeRateTable";

const testData = [
    ['USD', '1.5'],
    ['EUR', '2.0'],
    ['AUD', '1.1'],
    ['JPY', '0.03'],
]

describe('ExchangeRateTable test', () => {
    test('render correctly', () => {
        const { container } = render(<ExchangeRateTable exchangeRates={testData} />);
        expect(container.firstChild).toMatchSnapshot();
       // screen.debug();
        expect(screen.getByText('USD')).toBeInTheDocument();
        expect(screen.getByText('1.50')).toBeInTheDocument();
    })
})