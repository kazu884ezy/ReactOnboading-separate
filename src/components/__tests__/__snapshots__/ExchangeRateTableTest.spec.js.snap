// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRateTable test render correctly 1`] = `
<table
  class="MuiTable-root makeStyles-table-1"
>
  <thead
    class="MuiTableHead-root"
  >
    <tr
      class="MuiTableRow-root MuiTableRow-head"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Currency
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Value
      </th>
    </tr>
  </thead>
  <tbody
    class="MuiTableBody-root"
  >
    <tr
      class="MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        USD
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        1.50
      </td>
    </tr>
    <tr
      class="MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        EUR
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        2.00
      </td>
    </tr>
    <tr
      class="MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        AUD
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        1.10
      </td>
    </tr>
    <tr
      class="MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        JPY
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        0.03
      </td>
    </tr>
  </tbody>
</table>
`;
